src/themes/material/components/edit-chip-list/edit-chip-list.component.ts

```diff
@@ -0,0 +1,198 @@
+import {
+	Component,
+	Input,
+	OnInit,
+	ChangeDetectorRef,
+	ViewChildren,
+	QueryList
+} from '@angular/core';
+
+import {AppError} from 'ng2-qgrid/core/infrastructure';
+import {Command, CommandManager} from 'ng2-qgrid/core/command';
+import {Shortcut, ShortcutManager} from 'ng2-qgrid/core/shortcut';
+import {EditChipComponent} from './edit-chip/edit-chip.component';
+import {ChipState} from './edit-chip/edit-chip.component';
+
+interface ISelected {
+	index?: number;
+	value?: string;
+	action: Actions;
+}
+
+interface IChipItem {
+	value?: string | any;
```

Comment:
why u need this abstraction?

Reply:
The answer is similar to my previous comment. Furthermore, it seems to be harder to implement required functionality without this abstraction, considerably increasing the code complexity and reducing its elegance.
Secondly, I believe I will need this abstraction to add the TrackBy support to the component because it will be necessary to track additional property values of the item, not just the value to be rendered on the form.

Reply:
there can be `placeholder` abstraction to create new values but don't understand whe we need to wrap chip values, what state can have particular chip? 
consider for next composition:
1. one state for whole chip-list = 'edi | view'
2. selectionIndex property for whole chip-list (selectionIndex === -1 means placeholder is active)

Reply:
Regarding the TrackBy support: if I understand correctly this feature requires identifying the item so it can track its state changes. Just plain value is not enough as we agreed that is isn't necessarily unique. So, using this abstraction I can use the item index as an ID and track other property values, such as value, selected state, mode (view/edit) when they are changed

Reply:
FYI: the following are some of the useful resources I used to familiarize myself with the Track By:
- [How to use trackBy with ngFor](https://stackoverflow.com/questions/42108217/how-to-use-trackby-with-ngfor);
- [how to use track by inside ngFor angular 2](https://stackoverflow.com/questions/36322829/how-to-use-track-by-inside-ngfor-angular-2).